monarchmoney/monarchmoney.py

```diff
@@ -2395,6 +2377,95 @@ async def upload_account_balance_history(
             if resp.status != 200:
                 raise RequestFailedException(f"HTTP Code {resp.status}: {resp.reason}")
 
+    async def get_recurring_transactions(
+        self, start_date: str, end_date: str
+    ) -> Dict[str, Any]:
+        """
+        Fetches upcoming recurring transactions from Monarch Money's API.  This includes
+        all merchant data, as well as the accounts where the charge will take place.
+        """
+        query = gql(
+            """
+            query Web_GetUpcomingRecurringTransactionItems($startDate: Date!, $endDate: Date!, $filters: RecurringTransactionFilter) {
+              recurringTransactionItems(
+                startDate: $startDate
+                endDate: $endDate
+                filters: $filters
+              ) {
+                stream {
+                  id
+                  frequency
+                  amount
+                  isApproximate
+                  merchant {
+                    id
+                    name
+                    logoUrl
+                    __typename
+                  }
+                  __typename
+                }
+                date
+                isPast
+                transactionId
+                amount
+                amountDiff
+                category {
+                  id
+                  name
+                  icon
+                  __typename
+                }
+                account {
+                  id
+                  displayName
+                  icon
+                  logoUrl
+                  __typename
+                }
+                __typename
+              }
+            }
+        """
+        )
+
+        variables = {"startDate": start_date, "endDate": end_date}
+
+        if start_date is None ^ end_date is None:
```

Comment:
This is also producing an error for me... maybe it's my python version or something funky w/ `ipython`.

```
In [7]: result = asyncio.run(mm.get_recurring_transactions("2024-02-01", "2024-02-29"))
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[7], line 1
----> 1 result = asyncio.run(mm.get_recurring_transactions("2024-02-01", "2024-02-29"))

File /opt/homebrew/Cellar/python@3.11/3.11.6_1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/runners.py:190, in run(main, debug)
    186     raise RuntimeError(
    187         "asyncio.run() cannot be called from a running event loop")
    189 with Runner(debug=debug) as runner:
--> 190     return runner.run(main)

File /opt/homebrew/Cellar/python@3.11/3.11.6_1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/runners.py:118, in Runner.run(self, coro, context)
    116 self._interrupt_count = 0
    117 try:
--> 118     return self._loop.run_until_complete(task)
    119 except exceptions.CancelledError:
    120     if self._interrupt_count > 0:

File /opt/homebrew/Cellar/python@3.11/3.11.6_1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py:653, in BaseEventLoop.run_until_complete(self, future)
    650 if not future.done():
    651     raise RuntimeError('Event loop stopped before Future completed.')
--> 653 return future.result()

File ~/workspaces/github/monarchmoney/monarchmoney/monarchmoney.py:2434, in MonarchMoney.get_recurring_transactions(self, start_date, end_date)
   2387 query = gql(
   2388     """
   2389     query Web_GetUpcomingRecurringTransactionItems($startDate: Date!, $endDate: Date!, $filters: RecurringTransactionFilter) {
   (...)
   2429 """
   2430 )
   2432 variables = {"startDate": start_date, "endDate": end_date}
-> 2434 if start_date is None ^ end_date is None:
   2435     raise Exception(
   2436         "You must specify both a start_date and end_date, not just one of them."
   2437     )
   2438 elif start_date is None and end_date is None:

TypeError: unsupported operand type(s) for ^: 'NoneType' and 'str'
```

In the above example, the following snippet of line 2434 is highlighted: `None ^ end_date`, suggesting it's being interpreted before the left/right conditions.

For me, this error goes away when I surround both conditions in parentheses (i.e. `(start_date is None) ^ (end_date is None)`)

Reply:
oof, yep, it's the order of operations comparing `None` to `end_date`.  fixing...